lib/wick-get-iface-ip

```diff
@@ -20,36 +20,58 @@
 #
 # Returns 0 on success, non-zero on failure.
 wickGetIfaceIp() {
-    local ip list
+    local list target interface allInterfaces
 
-    if [[ "$2" == '*' ]]; then
-        list=()
+    wickGetArgument target 0 "$@"
+    wickGetArgument interface 1 "$@"
 
-        while read ip; do
-            ip=${ip#*inet addr:}
-            ip=${ip#*inet }
-            ip=${ip%% *}
-            list[${#list[@]}]=$ip
-        done < <(ifconfig -v 2>/dev/null | grep 'inet ' || echo "")
+    list=()
+    allInterfaces=false
 
+    if [[ "$interface" == "*" ]]; then
+        interface=""
+        allInterfaces=true
+    fi
+
+    if wickCommandExists ip; then
+        cmd=(ip addr show)
+
+        if [[ -n "$interface" ]]; then
+            cmd[${#cmd[@]}]="$interface"
+        fi
+
+        list=( $("${cmd[@]}" 2>/dev/null | grep -Po "inet\s+\K[\d.]+") )
+    elif wickCommandExists ifconfig; then
+        cmd=(ifconfig -v)
+
+        if [[ -n "$interface" ]]; then
+            cmd[${#cmd[@]}]="$interface"
+        fi
+
+        list=( $("${cmd[@]}" 2>/dev/null | grep -Po "inet\s+(addr:)?\K[\d.]+") )
+    else
+        wickError "Couldn't find a command to look up ip addresses"
+
+        return 2
+    fi
+
+    if [[ "$allInterfaces" == true ]]; then
         if [[ ${#list[@]} -eq 0 ]]; then
-            local "$1" && wickIndirectArray "$1"
+            local "$target" && wickIndirectArray "$target"
```

Comment:
I currently do the magic-like check.  What I would like to do is this:

    myArray=()
    local "$dest" && wickIndirectArray "$dest" "${myArray[@]-?}"

However this sends an empty string if the array is empty.

    local "$dest" && wickIndirectByRef "$dest" "myArray"

Well, I will document this and put it in a ticket.  Consider this resolved.